rpc/src/p2p.rs

```diff
@@ -1,17 +1,207 @@
 use jsonrpsee::proc_macros::rpc;
-use serde::{Deserialize, Serialize};
+use libp2p::Multiaddr;
+use serde::{de, de::Error, ser, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::str::FromStr;
+use thiserror::Error;
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct AddrInfo {
     #[serde(rename = "ID")]
-    pub id: String,
-    // TODO: multiaddr
+    pub id: PeerId,
     #[serde(rename = "Addrs")]
-    pub addrs: Vec<String>,
+    pub addrs: Vec<Multiaddr>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Stat {
```

Comment:
type definitions should be in the `type` crate

Reply:
moved all the new types to [`types`](https://github.com/eigerco/celestia-node-rs/pull/52/files#diff-ed39cfb7fdc8b39ea00751ee4b418a96975ad12b235bc308b96be3246a2c8686).

Reply:
hmm, it's arguable if `celestia-types` is a place for the p2p related celestia types. I was previously more thinking about it as of kind of 'celestia-native' types, like eg. EH. I'm not sure I like that the celestia-types will depend also on the `libp2p`. If those types are needed only for the rpc I like them there, or in future closer to celestia-node if it's gonna mimic the json-rpc server

Reply:
i think keeping them in `celestia-types` and feature guarded with `libp2p` is a good solution for now